core-java/src/test/java/org/baeldung/java/io/JavaFilePathUnitTest.java

```diff
@@ -0,0 +1,97 @@
+package org.baeldung.java.io;
+
+import static org.junit.Assert.*;
+
+import org.junit.AfterClass;
+import org.junit.BeforeClass;
+import org.junit.Test;
+
+import java.io.File;
+import java.io.IOException;
+
+public class JavaFilePathUnitTest {
+
+    private static String userDir;
+
+    @BeforeClass
+    public static void createFilesAndFolders() throws IOException {
+        userDir = System.getProperty("user.dir");
```

Comment:
Instead of the user's home directory, make use of the TemporaryFolder construct provided in JUnit Rules.